### ISSUE TITLE
Nested array/hash layout misalignment

### ISSUE THREAD

**Issue by Ardena** [rubocop/rubocop#14547]
<img width="700" height="569" alt="Image" src="https://github.com/user-attachments/assets/911cdb8d-00d3-4f78-b584-66e6d6cc91c3" />

I noticed this line looked super weird and wasn't aligned, so I fixed it.  When I ran rubocop, it changed it back to look like this.

--------

## Expected behavior

The line in the hash and the next hash line should align with the others in the array.

## Steps to reproduce the problem

Hashes in an array inside of a hash seem to have issues

## RuboCop version

1.80.0 (using Parser 3.3.9.0, Prism 1.4.0, rubocop-ast 1.46.0, analyzing as Ruby 3.4, running on ruby 3.4.1) [arm64-darwin24]
  - rubocop-factory_bot 2.27.1
  - rubocop-graphql 1.5.6
  - rubocop-rails 2.33.3
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.5.0
  - rubocop-rspec_rails 2.31.0
  - rubocop-performance 1.25.0

**Comment by ydakuka** [comment]
This can be achieved by specifying the following in `.rubocop.yml`:
https://docs.rubocop.org/rubocop/cops_layout.html#layoutfirsthashelementindentation

```yaml
Layout/FirstHashElementIndentation:
  EnforcedStyle: consistent
```

**Comment by bbatsov** [comment]
Small correction to the above: the cop driving this is `Layout/ArrayAlignment`, not `Layout/FirstHashElementIndentation`. Its default `with_first_element` style aligns the second element under the first (right after the `[`), which is where the deep indent comes from. Setting `Layout/ArrayAlignment: EnforcedStyle: with_fixed_indentation` gives you the one-level indent instead. Closing as a config question - happy to reopen if that doesn't do it.
